fabric-textures-v0/src/main/java/net/fabricmc/fabric/api/client/texture/ColorSniffer.java

```diff
@@ -0,0 +1,52 @@
+package net.fabricmc.fabric.impl.client.texture;
+
+import net.fabricmc.fabric.mixin.client.texture.SpriteAccessor;
+import net.minecraft.client.texture.NativeImage;
+import net.minecraft.client.texture.Sprite;
+
+/**
+ * Utility class for averaging Sprite colors.
+ * You should call one of these methods!
+ */
+public class ColorSniffer {
+
+	/**
+	 * @param sprite The Sprite you want to get the average color from.
+	 * @return The average ABGR color of the provided Sprite.
+	 */
+	public static int getAverageSpriteColor(Sprite sprite) { return getAverageSpriteColor(sprite, 0); }
+
+	/**
+	 * @param sprite The Sprite you want to get the average color from.
+	 * @param frame The specific frame of the sprite, in case it has more than one.
+	 * @return The average ABGR color of the provided Sprite, during the provided frame.
+	 */
+	public static int getAverageSpriteColor(Sprite sprite, int frame) {
+		if (sprite == null) {
+			return 0xFFFF00FF;
+		}
+
+		NativeImage image = ((SpriteAccessor) sprite).getImages()[frame];
+		int pixelCount = 0;
+		long avgR = 0, avgG = 0, avgB = 0;
+		for (int y = 0; y < sprite.getHeight(); y++) {
+			for (int x = 0; x < sprite.getWidth(); x++) {
+				int c = image.getPixelRgba(x, y);
+				if (((c >> 24) & 0xFF) != 0x00) {
+					avgB += ((c >> 16) & 0xFF) * ((c >> 16) & 0xFF);
```

Comment:
Maybe write a utility method like `sqr`, and do `avgB += sqr((c >> 16) & 0xFF)` instead